kernel/byterun/coq_interp.c

```diff
@@ -1222,8 +1222,10 @@ value coq_interprete
         CheckInt2();
 	/* returns the sum with a carry */
         value s;
-        s = uint63_add(accu, *sp++);
-        AllocCarry(uint63_lt(s,accu));
+        int cond;
+        Uint63_add(s, accu, *sp++);
+        Uint63_lt(cond,s,accu);
+        AllocCarry(cond);
```

Comment:
`s` might die there.

In fact, I think it is error-prone not to use `accu` as the implicit return value of all these macros. Had you done so (and the original code too), you would have been forced to preserve it.

Reply:
Indeed, thanks for the advice, I followed it.